### PR TITLE
Test PFS integration refactoring

### LINKED ISSUE #4565: Refactor PFS interaction unit tests for clarity
The current PFS interaction unit tests are hard to understand. We should refactor them.

### PR DESCRIPTION
**Pull request by Dominik1999**
partially closes #4565

### CODE REVIEW

File: raiden/network/pathfinding.py [hunk 1]
```diff
@@ -148,8 +148,8 @@ def get_pfs_info(url: str) -> Optional[PFSInfo]:
             operator=infos["operator"],
             version=infos["version"],
         )
-    except (json.JSONDecodeError, requests.exceptions.RequestException, KeyError):
-        return None
+    except (json.JSONDecodeError, requests.exceptions.RequestException, KeyError) as e:
+        raise ServiceRequestFailed(e)
```
Comment (palango) [inline raiden/network/pathfinding.py]: This should be `raise ServiceRequestFailed(e.message) from e` (and we have to hope that all of the exceptions have a message, but that's why we have tests...)

Comment (Dominik1999) [reply raiden/network/pathfinding.py]: `AttributeError: 'JSONDecodeError' object has no attribute 'value'` nor
AttributeError: 'JSONDecodeError' object has no attribute 'message'

Comment (palango) [reply raiden/network/pathfinding.py]: Then maybe `raise ServiceRequestFailed(str(e)) from e`

File: raiden/network/pathfinding.py [hunk 2]
```diff
@@ -256,11 +257,12 @@ def configure_pfs_or_exit(
...
 
-    pathfinding_service_info = get_pfs_info(pfs_url)
-    if not pathfinding_service_info:
+    try:
+        pathfinding_service_info = get_pfs_info(pfs_url)
+    except ServiceRequestFailed as e:
         click.secho(
             f"There is an error with the pathfinding service with address "
-            f"{pfs_url}. Raiden will shut down."
+            f"{pfs_url}. {e}. Raiden will shut down."
```
Comment (palango) [inline raiden/network/pathfinding.py]: I'd add something like `Error message: str(e)`

File: raiden/tests/unit/test_pfs_integration_refactored.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import pytest
+import requests
+from eth_utils import to_canonical_address
+
+from raiden.exceptions import ServiceRequestFailed
+from raiden.network.pathfinding import PFSInfo, get_pfs_info
+
+# We first test the correct handling of the pfs info endpoint. The info endpoint provides
+# the Raiden Client with price and information about the token network registry.
+# The client should handle incorrect information and format
```
Comment (palango) [inline raiden/tests/unit/test_pfs_integration_refactored.py]: format -> formatting

File: raiden/tests/unit/test_pfs_integration_refactored.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+
+# this test tests the correct handling of the success case
+def test_get_pfs_info_success():
+
+    info_data = {
+        "price_info": 5,
+        "network_info": {
+            "chain_id": 42,
+            "registry_address": "0xB9633dd9a9a71F22C933bF121d7a22008f66B908",
```
Comment (palango) [inline raiden/tests/unit/test_pfs_integration_refactored.py]: maybe make this a variable above and use here and below.

File: raiden/tests/unit/test_startup.py [hunk 5]
```diff
@@ -179,7 +201,7 @@ def test_setup_proxies_all_addresses_are_given():
     contracts = {}
     blockchain_service = MockChain(network_id=network_id, node_address=make_address())
 
-    with patched_get_for_succesful_pfs_info():
+    with patch.object(pathfinding, "get_pfs_info", return_value=PFS_CONFIG.info):
```
Comment (palango) [inline raiden/tests/unit/test_startup.py]: If all you use is the `PFSInfo`, maybe just instantiate this above.

File: raiden/network/pathfinding.py (hunk 2 above)
Comment (Dominik1999) [reply raiden/network/pathfinding.py]: ok done

File: raiden/tests/unit/test_startup.py (hunk 5 above)
Comment (Dominik1999) [reply raiden/tests/unit/test_startup.py]: I might not understand that, will check

File: raiden/network/pathfinding.py (hunk 1 above)
Comment (Dominik1999) [reply raiden/network/pathfinding.py]: ok done

**Comment by palango** [comment]
The commit still needs a better title

**Pull request merged**